### PR TITLE
Feature/iteration 3

### PR DESCRIPTION
**Pull request by delilahrois**
**What (if any) features are you implementing?**
 -We are now successfully fetching data from the API to populate the DOM / update state.
 
**What (if anything) did you refactor?**
 -Refactored from hard data to use the fetch call for data in the API. 
 
**Were there any issues that arose?**
-No issues at this point. Fetch is fully functional and error handling is clean.

**Is there anything needs to be prioritized?**
 -Nope! Good to go for the weekend. We may want to add some styling to clean things up at this point.
 
**Is there anything that you need from your teammates?**
 -N/A

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,12 +8,25 @@ class App extends Component {
...
       singleMovie: {},
+      error: null
     }
   }
 
+  componentDidMount = () => {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+    .then(response => response.json())
+    .then(data => {
+      this.setState({ movies: data.movies })
```
Comment (michelecomfort) [inline src/App.js:22]: Good eye on using dot notation to pull the array from object!!!!!!!!!

File: src/App.js [hunk 2]
```diff
@@ -8,12 +8,25 @@ class App extends Component {
...
+    })
+    .catch(err => {
+      console.log(err)
+      this.setState({ error: err })
```
Comment (michelecomfort) [inline src/App.js:26]: I was just thinking that maybe if the server is down we render a different message that says "Try back later?" That way they don't keep refreshing if it's a server error? Let's brainstorm on how we want to handle this.

File: src/Page.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import './Page.css'
 
-const Page = ({movie, returnToMain}) => {
+const Page = ({ movie, returnToMain }) => {
```
Comment (michelecomfort) [inline src/Page.js:4]: Thank you for prettifying! (is that a legit word?)

**Review by michelecomfort — APPROVED** [review]
That was fast, efficient work today my friend! Kudos and well done!

**Comment by michelecomfort** [comment]
That was fast, efficient work today my friend! Kudos and well done!!

File: src/App.js [hunk 4]
```diff
@@ -8,12 +8,25 @@ class App extends Component {
...
-      movies: movieData.movies,
+      movies: [],
       onMainPage: true,
       singleMovie: {},
+      error: null
     }
   }
 
+  componentDidMount = () => {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (kaylagordon) [inline src/App.js:19]: consider moving fetch calls to their own file

File: src/App.js [hunk 5]
```diff
@@ -8,12 +8,25 @@ class App extends Component {
...
+    .then(response => response.json())
+    .then(data => {
+      this.setState({ movies: data.movies })
+    })
+    .catch(err => {
+      console.log(err)
```
Comment (kaylagordon) [inline src/App.js:25]: remove console logs

File: src/App.js [hunk 6]
```diff
@@ -24,19 +37,22 @@ class App extends Component {
   returnToMain = () => {
     this.setState({ onMainPage: true })
```
Comment (kaylagordon) [inline src/App.js:38]: You can cut this once you have Router implemented!

File: src/App.js [hunk 7]
```diff
@@ -24,19 +37,22 @@ class App extends Component {
...
   }
+
   render() {
     return (
     this.state.onMainPage ?
       (<main>
         <h1>Rancid Tomatillos</h1>
+        { this.state.error && <p>Oops! Something went wrong. Refresh and try again.</p> }
```
Comment (kaylagordon) [inline src/App.js:46]: What if you held that string  as the value of `this.state.error` instead of a boolean?

**Review by kaylagordon — COMMENTED** [review]
Everything looks great! No need to make any changes - these are all just considerations :)

File: src/App.js (hunk 4 above)
Comment (michelecomfort) [reply src/App.js:19]: Appreciate this idea!

File: src/App.js (hunk 5 above)
Comment (michelecomfort) [reply src/App.js:25]: YES, thank you!!!

File: src/App.js (hunk 7 above)
Comment (michelecomfort) [reply src/App.js:46]: Delilah - what are your thoughts on this? Would that be something like...
`{this.state.error && <p>{this.state.error}</p>}`
Am I interpreting this correctly?

**Pull request merged**